builder/ebssurrogate/builder_acc_test.go

```diff
@@ -0,0 +1,135 @@
+package ebssurrogate
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+	"time"
+
+	amazon_acc "github.com/hashicorp/packer-plugin-amazon/builder/ebs/acceptance"
+	"github.com/hashicorp/packer-plugin-sdk/acctest"
+)
+
+func TestAccBuilder_EbssurrogateBasic(t *testing.T) {
+	ami := amazon_acc.AMIHelper{
+		Region: "us-east-1",
+		Name:   fmt.Sprintf("packer-plugin-amazon-ebs-basic-acc-test %d", time.Now().Unix()),
+	}
+	testCase := &acctest.PluginTestCase{
+		Name:     "amazon-ebssurrogate_basic_test",
+		Template: fmt.Sprintf(testBuilderAccBasic, ami.Name),
+		Teardown: func() error {
+			return ami.CleanUpAmi()
+		},
+		Check: func(buildCommand *exec.Cmd, logfile string) error {
+			if buildCommand.ProcessState != nil {
+				if buildCommand.ProcessState.ExitCode() != 0 {
+					return fmt.Errorf("Bad exit code. Logfile: %s", logfile)
+				}
+			}
+			return nil
+		},
+	}
+	acctest.TestPlugin(t, testCase)
+}
+
+func TestAccBuilder_EbssurrogateBasic_forceIMDSv2(t *testing.T) {
+	aminame := fmt.Sprintf("packer-plugin-amazon-ebs-basic-acc-test %d", time.Now().Unix())
+	ami := amazon_acc.AMIHelper{
+		Region: "us-east-1",
+		Name:   aminame,
+	}
+	testCase := &acctest.PluginTestCase{
+		Name:     "amazon-ebssurrogate_basic_test_imdsv2",
+		Template: fmt.Sprintf(testBuilderAccBasicIMDSv2, aminame),
+		Teardown: func() error {
+			return ami.CleanUpAmi()
+		},
+		Check: func(buildCommand *exec.Cmd, logfile string) error {
+			if buildCommand.ProcessState != nil {
+				if buildCommand.ProcessState.ExitCode() != 0 {
+					return fmt.Errorf("Bad exit code. Logfile: %s", logfile)
+				}
+			}
+
+			ami := amazon_acc.AMIHelper{
+				Region: "us-east-1",
+				Name:   aminame,
+			}
```

Comment:
You make this call on line 38, as it's needed in the Teardown function. Can use reuse ami here? Or does you need to reinitialize?

Reply:
I'm not sure what you mean by "reuse the ami here", could you clarify? Both tests have their own AMI with a different name after integrating your last suggestions, anything else that you think could be relevant regarding those tests?

Reply:
OK my bad, I just noticed the duplicate. I've removed it, rerolling the commit